### PR TITLE
[W5.4][W10-A1] Ong Yong Siang

### PR DESCRIPTION
**Pull request by KeithSOC**
**[Add Sort Feature]**

1. SortCommand.java --> Add new COMMAND_WORD, Add new NameComparator method, Override with new execute method
2. Parser.java --> Add SortCommand into switch case for user input
3. UniquePersonList.java --> Add mustableListView method to return sortable List
4. Command.java --> Add method for display text for sort command after it is executed "X persons sorted!"
5. Messages.java --> Add display text for sort command after it is executed "X persons sorted!"
6. HelpCommand.java --> Add help message for the new sort command

**[Update User Guide]**

1. UserGuide.md

**[Update Test Files]**

1. expected.txt
2. input.txt --> Add sort after adding Esther and listing 5 persons listed

**[Add Eclipse project configuration files]**

1. .classpath
2. .project

### CODE REVIEW

File: .classpath [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (500poundbear) [inline .classpath]: Hi @KeithSOC, I don't think the .classpath file should be part of your pull request

File: .project [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (500poundbear) [inline .project]: Hi @KeithSOC, .project is project specific to your setup on Eclipse and shouldn't be part of your PR!

File: src/seedu/addressbook/commands/Command.java [hunk 3]
```diff
@@ -27,14 +28,18 @@ protected Command() {
     }
 
     /**
-     * Constructs a feedback message to summarise an operation that displayed a listing of persons.
+     * Constructs a feedback message to summarize an operation that displays a listing of persons.
```
Comment (500poundbear) [inline src/seedu/addressbook/commands/Command.java]: @KeithSOC Good catch on the British spelling. However the intention of this PR should be to propose a new feature `sort`. Your amendments here should be part of a separate PR.

File: src/seedu/addressbook/commands/SortCommand.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
+/**
+ * NUS School of Computing | Computer Science
+ * - - - - - - - - - - - - - - - -
+ * Name		 	: Ong Yong Siang
+ * Matric No.	: A0156051N
```
Comment (500poundbear) [inline src/seedu/addressbook/commands/SortCommand.java]: @KeithSOC I don't think this is needed because it is not part of the requirements. Additionally, git has this `blame` feature that can reveal to us who's the owner of a specific line of code in the file. So this should be removed!

File: test/expected.txt [hunk 5]
```diff
@@ -19,6 +19,8 @@
 || Example: find alice bob charlie
 || list: Displays all persons in the address book as a list with index numbers.
 || Example: list
+|| sort: Displays all persons in the address book as a sorted (A~Z in ascending order) list with index numbers.
+|| Example: sort
 || view: Views the non-private details of the person identified by the index number in the last shown person listing.
```
Comment (500poundbear) [inline test/expected.txt]: @KeithSOC Because the `persons` where added in order, I think what can be improved will be to add the 5 person with their names in an non-alphabetical order so that we can be more certain that the `sort` feature is doing what we expect it to do.

File: src/seedu/addressbook/commands/SortCommand.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+	class NameComparator implements Comparator<Person> {
+		public int compare(Person p1, Person p2) {
+			// Compares its two arguments for order by name
+			return p1.getName().toString().compareTo(p2.getName().toString());
+		}
+
+		public boolean equals(Object obj) {
+			// Simply checks to see if we have the same object
+			return this == obj;
+		}
```
Comment (500poundbear) [inline src/seedu/addressbook/commands/SortCommand.java]: @KeithSOC There is no need to implement the equals function here. Besides, objects shouldn't be compared by the `==` sign. Equality between 2 Person objects in this case should be defined by having totally matching attributes.

**Comment by KeithSOC** [comment]
@500poundbear Thank you very much for examining my codes thoroughly. Your comments are very constructive. Appreciate it! I will work on your suggestions and advice!

File: doc/UserGuide.md [hunk 7]
```diff
@@ -53,6 +53,10 @@ Examples:
 Shows a list of all persons in the address book.<br>
 Format: `list`
 
+### Listing all sorted persons : `sort`
+Shows a list of all sorted persons (A~Z in ascending order) in the address book.<br>
```
[the remark this answers is not in the document]
Comment (damithc) [reply doc/UserGuide.md]: Does this command simply shows a sorted list or permanently sorts the persons in the address book? If it is the latter, this description is incorrect.
